apps/emqx_bridge_kinesis/src/emqx_bridge_kinesis_impl_producer.erl

```diff
@@ -123,31 +119,111 @@ on_get_status(_InstanceId, #{pool_name := Pool} = State) ->
             disconnected
     end.
 
+on_add_channel(
+    _InstId,
+    #{
+        installed_channels := InstalledChannels
+    } = OldState,
+    ChannelId,
+    ChannelConfig
+) ->
+    {ok, ChannelState} = create_channel_state(ChannelConfig),
+    NewInstalledChannels = maps:put(ChannelId, ChannelState, InstalledChannels),
+    %% Update state
+    NewState = OldState#{installed_channels => NewInstalledChannels},
+    {ok, NewState}.
+
+create_channel_state(
+    #{parameters := Parameters} = _ChannelConfig
+) ->
+    #{
+        stream_name := StreamName,
+        partition_key := PartitionKey
+    } = Parameters,
+    {ok, #{
+        templates => parse_template(Parameters),
+        stream_name => StreamName,
+        partition_key => PartitionKey
+    }}.
+
+on_remove_channel(
+    _InstId,
+    #{
+        installed_channels := InstalledChannels
+    } = OldState,
+    ChannelId
+) ->
+    NewInstalledChannels = maps:remove(ChannelId, InstalledChannels),
+    %% Update state
+    NewState = OldState#{installed_channels => NewInstalledChannels},
+    {ok, NewState}.
+
+on_get_channel_status(
+    _ResId,
+    ChannelId,
+    #{
+        pool_name := PoolName,
+        installed_channels := Channels
+    } = State
+) ->
+    #{stream_name := StreamName} = maps:get(ChannelId, Channels),
+    case
+        emqx_resource_pool:health_check_workers(
+            PoolName,
+            {emqx_bridge_kinesis_connector_client, connection_status, [StreamName]},
+            ?HEALTH_CHECK_TIMEOUT,
+            #{return_values => true}
+        )
+    of
+        {ok, Values} ->
+            AllOk = lists:all(fun(S) -> S =:= {ok, connected} end, Values),
```

Comment:
Please use `?status_connected`, `?status_disconnected`, etc. from `emqx_resource.hrl` here and in other places in this file, so we may avoid typos and normalize its usage.

Reply:
Sure, will fix. This code is basically the same as it was before the refactoring.

Reply:
Fixed

Reply:
Fixed

Reply:
Thanks!  Since we're refactoring here, it seems a good time to fix this bridge. :beers: